### PR TITLE
Bunch of fixes for `dub test`

### PR DESCRIPTION
**Pull request by WebFreak001**
I tried to build and test the project, but there were a bunch of issues.

I cleaned up the code according to D style, removed some unnecessary code and made the parsing more explicit.

I recommend going through the commits one-by-one.

If anything, I think at least 93b86675cf7606de80cbb71f472e9f4a11bbb4e4 is a must-have part of this PR, so if you decide against merging, please consider adding the changes from that. (It adds `.read!"property"` instead of the `.property` access to QuicReader, which tells the user that decoding is happening and that you can't/shouldn't access a property twice for proper reading)

I did not really read the QUIC RFC for these changes. There are definitely breaking changes in this PR. All the tests pass locally, so there should definitely be a bunch more tests based on the RFC.

I based my work on the devel-session branch, because that was most recently linked on the D forum and in the branches view it has the most work. I think you should definitely merge it into main, as main currently has no actual code in it and makes it hard to discover what's actually being done here.

### CODE REVIEW

File: src/quic/frame_writer.d [hunk 1]
```diff
@@ -114,5 +114,5 @@ unittest
     LocalAppender buffer;
     writer.getBytes(buffer, packet);
     assert(buffer[].toHexString!(LetterCase.lower) ==
-                                "c00000000108000102030405060705635f63696400");
+                                "c00000000108000102030405060705635f6369640000");
```
Comment (vladchicos) [inline src/quic/frame_writer.d:117]: Thanks! You are right, your test is better. The additional 0 byte would mean that packetPayload is empty.

File: src/quic/frame_reader.d [hunk 2]
```diff
@@ -83,6 +84,11 @@ struct QuicReader(FrameType)
                 return networkStream[bufIndex-len..bufIndex];
             }
 
+            else static if (isIntegral!FieldType)
+            {
+                return readBigEndianField(networkStream, bufIndex, FieldType.sizeof);
```
Comment (vladchicos) [inline src/quic/frame_reader.d:89]: It is the expected behaviour. Useful in tls framing, where there are no VarInts, and the fields without UDAs have a fixed, already known size. You can look at https://www.rfc-editor.org/rfc/rfc8446.html#page-28 for reference. I did something similar in QuicWriter, although I used isUnsigned (there are no signed fields).

File: src/quic/frame_reader.d [hunk 3]
```diff
@@ -45,32 +58,38 @@ struct QuicReader(FrameType)
 
             else static if (is(FieldType == ubyte[]))
             {
-                alias attributes = __traits(getAttributes, mixin(FrameType.stringof, '.', name)); 
-                static if(is(attributes == VarIntLength))
```
Comment (vladchicos) [inline src/quic/frame_reader.d:49]: This was the source of those `Error: no property packetPayload` errors.

**Comment by vladchicos** [comment]
Hi @WebFreak001! Thanks for your help. I went through your commits and I think that your changes are great. Indeed, my code did not check if the user tries to add UDAs to the fields/frames that do not require additional information to be decoded/encoded. I do agree with you that more tests based on the RFCs should be added.

**Comment by vladchicos** [comment]
@WebFreak001 In this devel-session branch I also merged the code from https://github.com/dlang-community/quic-d/tree/devel and https://github.com/dlang-community/quic-d/tree/devel-frame (as you have seen, QuicSession needs those modules). What should I do with those branches?

**Comment by WebFreak001** [comment]
I would recommend deleting branches once they are merged and finished and just having main to work on.

For larger features changing a bunch of things having another branch is fine too though.

Large projects also sometimes have two branches: one that you regularly work on and another one that just keeps getting merged into from your work branch, which just gets updated when releasing new versions or hot-fixing old releases and fixing regressions.

**Pull request merged**